tools/macos/README.md

```diff
@@ -1,8 +1,89 @@
-#docker-machine:
+# How to setup a Mac host for OpenWhisk
 
-* one time setup
+One way to develop or deploy OpenWhisk on a Mac is to use [docker-machine](https://docs.docker.com/machine/install-machine/) which runs the Docker daemon inside a virtual machine accessible from the Mac host.
+
+# Prerequisites
+
+The following are required to build and deploy OpenWhisk from a Mac host:
+
+- [Oracle VM VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+- [Docker](https://docs.docker.com/engine/installation/mac/) (including `docker-machine`)
+- [Java 8](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+- [Scala 2.11](http://scala-lang.org/download/)
+- [Ant](http://ant.apache.org/manual/install.html)
+- [Ant-Contrib Tasks](http://ant-contrib.sourceforge.net/)
+- [Gradle](https://docs.gradle.org/current/userguide/installation.html)
+
+[Homebrew](http://brew.sh/) is an easy way to install all of these and prepare your Mac to build and deploy OpenWhik. The following shell command is provided for your convenience to install `brew` with [Cask](https://github.com/caskroom/homebrew-cask) and bootstraps these to complete the setup. Copy the entire section below and paste it into your terminal to run it.
+
+```
+echo '
+# install homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+# install cask
+brew tap caskroom/cask
+# install virtualbox
+brew cask install virtualbox
+# install docker
+brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e010e91a6bb5ad6697fb158017b36f575bbe644f/Formula/docker.rb
+# install docker-machine
+brew install docker-machine
+# install java 8, scala
+brew cask install java
+brew install scala
+# install ant and gradle
+brew install ant
+brew install ant-contrib
+brew install gradle
+# the following is required for running tests
+sudo pip install jsonschema' | bash
+```
+
+# Create and configure Docker machine
+
+It is recommended that you create a virtual machine with at least 4GB of RAM.
+
+```
+docker-machine create -d virtualbox \
+   --virtualbox-memory 4096 \
+   --virtualbox-boot2docker-url=https://github.com/boot2docker/boot2docker/releases/download/v1.9.1/boot2docker.iso \
+    whisk # the name of your docker machine
+```
+
+The Docker virtual machine requires some tweaking to work from the Mac host with OpenWhisk.
+The following [script](./tweak-dockermachine.sh) will disable TLS, add port forwarding
+within the VM and routes `172.17.x.x` from the Mac host to the Docker virtual machine.
+Enter your sudo Mac password when prompted.
+
+```
+cd /your/path/to/openwhisk
+./tools/macos/tweak-dockermachine.sh
```

Comment:
Running this script in my setup leaves `docker-machine ls` output with error output about port 2376, should I ignore?
~/dev/whisk/git/openwhisk (master)
$ ./tools/macos/tweak-dockermachine.sh
bootsync.sh                                                                           100%   61     0.1KB/s   00:00
python.tcz.dep OK
tk.tcz.dep OK
readline.tcz.dep OK
Connecting to repo.tinycorelinux.net (89.22.99.37:80)
...
add net 172.17.0.0: gateway 192.168.99.100
172.17             192.168.99.100     UGSc            1        0 vboxnet
Run the following:
   export DOCKER_HOST=tcp://192.168.99.100:4243
~/dev/whisk/git/openwhisk (master)
$ export DOCKER_HOST=tcp://192.168.99.100:4243
~/dev/whisk/git/openwhisk (master)
$ docker-machine ls
NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER    ERRORS
default   -        virtualbox   Stopped                                       Unknown
whisk     -        virtualbox   Running   tcp://192.168.99.100:2376           Unknown   Unable to query docker version: Cannot connect to the docker engine endpoint